src/shallow.ts

```diff
@@ -10,6 +10,29 @@ function shallow<T>(objA: T, objB: T) {
   ) {
     return false
   }
+
+  if (objA instanceof Map && objB instanceof Map) {
+    if (objA.size !== objB.size) return false
+
+    for (const [key, value] of objA) {
+      if (value !== objB.get(key)) {
```

Comment:
Let's make it consistent:
```suggestion
      if (!Object.is(value, objB.get(key))) {
```